base/loading.jl

```diff
@@ -66,40 +67,37 @@ function _require_from_serialized(node::Int, path_to_try::ByteString, toplevel_l
         else
             content = remotecall_fetch(node, open, readbytes, path_to_try)
         end
-        if _include_from_serialized(content)
+        restored = _include_from_serialized(content)
+        if restored != nothing
```

Comment:
Oh, no, I see; the runtime doesn't realize this, because it needs to do a runtime type-check of the `Any` argument followed by a dynamic method dispatch.

However, I still think a separate PR that changes all of these `== nothing` comparisons in one shot is a better approach.   (In this particular PR the performance impact is negligible of course.)